test/unit/test_program.py

```diff
@@ -190,3 +191,28 @@ def test_compatibility_layer():
         print("type(original):", type(original))
         assert isinstance(transformed, type(original))
         assert transformed == original
+
+    print("=================\nOK\n=====================")
+    defmeascal = DefMeasureCalibration(
+        Qubit(0),
+        MemoryReference("foo"),
+        instructions
+    )
+    for original, transformed in zip(instructions, defmeascal.instrs):
+        assert isinstance(transformed, AbstractInstruction)
+        print("type(transformed):", type(transformed))
+        print("type(original):", type(original))
+        assert isinstance(transformed, type(original))
+        assert transformed == original
+
+
+def test_delay_conversions():
+    f = Frame([Qubit(0)], "hi")
+
+    i = DELAY(f, 7)
+    assert isinstance(i, DelayFrames)
+    p = Program()
+    p += i
+    p += DelayFrames([f], 8)
+    for x in p:
+        assert isinstance(x, DelayFrames), type(x)
```

Comment:
This test was passing previously, which makes me wonder what the problem is with the existing implementation on your side. Is there a small motivating example you can use to illustrate?

I.e. if `DelayFrames` recognized as `DelayFrames` even if it's technically a `Delay`, does that break downstream code? Are there places where we want to check if an instruction is exactly `Delay`?

Reply:
Ah, gotcha, the Qubit is what was missing, thanks for clarifying.